skonaki.py

```diff
@@ -159,16 +176,31 @@ def generate_summary(
         audio_size = audio.stat().st_size
     print(f"Audio file size in MB: {audio_size / 1000000}")
 
-    openai.api_key = api_key
-    print("Transcribing using OpenAI's Whisper")
-    with open(audio, "rb") as f:
-        transcript = openai.Audio.transcribe(
-            "whisper-1",
-            f,
-            response_format="srt",
-            language=language,
-            prompt=transcription_prompt,
-        )
+    if use_local_whisper:
+        print("Transcribing using local whisper")
+        local_whisper_model = whisper.load_model("base")
+        loaded_audio = whisper.load_audio(audio)
+        result = whisper.transcribe(model=local_whisper_model, audio=loaded_audio, language=language, prompt=transcription_prompt) 
+        # Need to use the get_writer() to get the output into srt format
+        # https://github.com/openai/whisper/discussions/758
+        writer = whisper.utils.get_writer("srt", ".")
+        # "None" set for options following answer here: https://github.com/openai/whisper/discussions/1229#discussioncomment-7091769
+        writer(result, audio, {"max_line_width":None, "max_line_count":None, "highlight_words":False})
+        # The writer() saves the file as audio.srt, and so the following
+        # lines are used to read the file into a string.
+        with open("audio.srt", "r") as f:
+            transcript = f.read()
```

Comment:
Have you checked out [this way](https://gist.github.com/yasinuygun/a5aaa24af9f5531872e4c6c863fd793f) of doing it? Seems like it's a bit neater.

Reply:
This definitely looks cleaner! However, I think the whisper library no longer has this function based on this commit: https://github.com/openai/whisper/commit/da600abd2b296a5450770b872c3765d0a5a5c769

two suggestions I have are:
1) Downgrade the whisper package and use write_srt()
2) I could move the current code into its own function as a way to make it cleaner with the current way

Reply:
Let's not use deprecated stuff, sounds good as it is.